libparsec/src/recovery.rs

```diff
@@ -0,0 +1,58 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+use std::sync::Arc;
+
+use libparsec_client::{create_device_from_recovery, ProxyConfig};
+pub use libparsec_client::{ExportRecoveryDeviceError, ImportRecoveryDeviceError};
+use libparsec_client_connection::AuthenticatedCmds;
+use libparsec_types::DeviceSaveStrategy;
+use libparsec_types::{AvailableDevice, DeviceLabel};
+
+use crate::handle::{borrow_from_handle, HandleItem};
+use crate::ClientConfig;
+use crate::Handle;
+
+pub async fn import_recovery_device(
+    config: ClientConfig,
+    recovery_device: Vec<u8>,
+    passphrase: String,
+    device_label: DeviceLabel,
+    save_strategy: DeviceSaveStrategy,
+) -> Result<AvailableDevice, ImportRecoveryDeviceError> {
+    let recovery_device: Arc<_> = libparsec_platform_device_loader::import_recovery_device(
```

Comment:
`libparsec` main crate is not convenient to test, so it should only contain the public API that is used for the javascript bindings (and also a bit of wrappers code to deal with the handle system)

So here you should put this `import_recovery_device` function in the `libparsec_client` crate (where it can easily be tested), then re-export it in `libparsec` (possibly creating a thin wrapper in `libparsec` if you need to deal with handles)